packages/web/components/stepper/index.tsx

```diff
@@ -0,0 +1,323 @@
+  return (
+    <IconButton
+      aria-label="Next step"
+      mode="unstyled"
+      icon={<Icon id="chevron-right" />}
+      className={classNames(
+        "h-3 w-3 text-osmoverse-400 disabled:cursor-not-allowed disabled:opacity-60",
+        className
+      )}
+      onClick={() => {
+        nextStep();
+      }}
+      disabled={activeStep === totalSteps - 1}
+    />
+  );
+};
+
+export const StepperLeftChevronNavigation: FunctionComponent<{
+  className?: string;
+}> = ({ className }) => {
+  const { previousStep, activeStep } = useStepperContext();
+  return (
+    <IconButton
+      aria-label="Previous step"
+      mode="unstyled"
+      icon={<Icon id="chevron-left" />}
+      className={classNames(
+        "h-3 w-3 text-osmoverse-400 disabled:cursor-not-allowed disabled:opacity-60",
+        className
+      )}
+      onClick={() => {
+        previousStep();
+      }}
+      disabled={activeStep === 0}
+    />
+  );
+};
+
+/**
+ * The Stepper component is a UI element that can be used to display content sequentially.
+ * It is an ideal choice for constructing features like slideshows, reels, or wizards where
+ * step-by-step navigation is required.
+ *
+ * It includes an autoplay feature, which allows automatic progression to the
+ * next step after a specified delay, providing a smooth, hands-free navigation experience.
+ *
+ * The Stepper is a headless component and requires multiple children to work:
+ *  - the `Step` serves to identify and render individual steps within the Stepper.
+ *    Each instance of Step corresponds to one stage in the sequential navigation.
+ *  - the `StepsIndicator` is responsible for generating a series of navigation buttons.
+ *    These buttons allow users to manually move between steps, granting a high level of navigational control.
+ *
+ * @example
+ * <Stepper autoplay={{ delayInMs: 4000, stopOnHover: true }}>
+ *  <Step>
+ *   <h4>Heading</h4>
+ *   <div className="h-12 w-12 bg-osmoverse-500"></div>
+ *   <p>test 1</p>
+ *  </Step>
+ *  <Step>
+ *   <p>test 2</p>
+ *  </Step>
+ *  <StepsIndicator />
+ * </Stepper>
+ */
+const Stepper: FunctionComponent<StepsProps> = (props) => {
+  const { children, autoplay } = props;
+
+  const timerTimeInMs = useRef(0);
+
+  const stepElements = Children.toArray(children).filter((child) => {
+    if (!child) return false;
+    return (child as ReactElement)?.props?.__TYPE === "Step";
+  });
+
+  const otherElements = Children.toArray(children).filter((child) => {
+    if (!child) return false;
+    return (child as ReactElement)?.props?.__TYPE !== "Step";
+  });
+
+  const stepsContext = useSteps({ count: stepElements.length });
+  const [isHovering, setIsHovering] = useState(false);
+
+  const isStopped = useMemo(
+    () => (autoplay?.stopOnHover && isHovering) || Boolean(autoplay?.isStopped),
+    [autoplay?.isStopped, autoplay?.stopOnHover, isHovering]
+  );
+
+  useEffect(() => {
+    if (autoplay && Boolean(autoplay?.delayInMs)) {
```

Comment:
This makes sense to me, but we could also use a setTimeout which might be less complex than using a counter.

Something like: 

```
useEffect(() => {
  if (autoplay && !isStopped) {
    // Schedule the next step to occur after the specified delay.
    const timeoutId = setTimeout(stepsContext.nextStep, autoplay.delayInMs);

    // The cleanup function cancels the timeout, if it hasn't already fired.
    return () => clearTimeout(timeoutId);
  }
}, [autoplay, stepsContext, isStopped]);
```

Just a suggestion

Reply:
@JoseRFelix can you address this please.